intake/source/cache.py

```diff
@@ -139,13 +141,20 @@ def load(self, urlpath):
                 logger.debug("Cached at: {}".format(cache_path))
                 self._log_metadata(urlpath, file_in.path, cache_path)
 
-                with file_in as f1:
-                    with file_out as f2:
-                        data = True
-                        while data:
-                            #TODO: print out progress
-                            data = f1.read(self.blocksize)
-                            f2.write(data)
+                file_size = file_in.fs.size(file_in.path)
```

Comment:
This will fail for HTTP

Reply:
No, HTTP implements `.size()`

Reply:
It does, but the server may well not supply the size when asked. If you look at the implementation, the behaviour is to try to cope with unknown sizes in a number of cases (particularly small files), but there is no general solution: HTTP is not FTP.

Reply:
Or are you talking about the [ValueError](https://github.com/dask/dask/blob/master/dask/bytes/http.py#L312)?  I can catch that.

Reply:
So, we ask for it and if the server doesn't provide it then we just don't show the progress bar.

Reply:
Yes.
Also the `.raise_for_status()` which can return any number of `requests` exceptions such as "I don't implement HEAD"; at the moment, those lower-level errors prevent any read, but they should be treated the same.